core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournalCheckpointThread.java

```diff
@@ -371,13 +379,23 @@ private void writeCheckpoint(long nextSequenceNumber) {
           nextSequenceNumber);
       mNextSequenceNumberToCheckpoint = nextSequenceNumber;
     } catch (IOException e) {
+      // IOException here means either failing to create UfsWriter
+      // or failing to close JournalWriter
       LOG.error("{}: Failed to checkpoint.", mMaster.getName(), e);
     }
   }
 
   @Override
   public void onError(Throwable t) {
-    setError(t);
+    if (ExceptionUtils.containsInterruptedException(t)) {
+      // Tolerate interruption when the master is failing over or closing
+      // so we don't extra-crash
+      LOG.warn("Thread {} interrupted, assume the master is failing over or shutting down",
+          Thread.currentThread().getId());
+    } else {
+      LOG.error("Uncaught exception from thread {}", Thread.currentThread().getId(), t);
```

Comment:
no crash here?

Reply:
yea seems to me the existing code here doesn't like to crash and relies on the `mTailerThread.awaitTermination()` to panic from the controller thread(probably the `main` thread)